### PR TITLE
React Group project 

### PR DESCRIPTION
**Pull request by ghreza-crypto**
Project requirements
Config & basic setup
Create React App.
Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
Create routes and view components: rockets, missions, my profile. Add dragons [only if your team has 3 members]).
Use <NavLink /> for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
Create directories for all Redux state slice files:rockets, missions, dragons [only if your team has 3 members].
Redux: Fetch data and update Redux store
Upon first render fetch data from the SpaceX API endpoints:
Rockets: https://api.spacexdata.com/v4/rockets
Missions: https://api.spacexdata.com/v3/missions
Dragons: https://api.spacexdata.com/v4/dragons [only if your team has 3 members]
Once the data are fetched, dispatch an action to store the selected data in Redux store:
Rockets:
id
rocket_name
description
flickr_images
Missions:
mission_id
mission_name
description
Dragons [only if your team has 3 members]:
id
name
type
flickr_images NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation). NOTE: Rockets is the default view, so you must fetch rockets data when the application starts. However, the missions data should only be fetched (once) when a user navigates to the Missions section.
Render UI:lists
Use useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes. i.e.:
// get rockets data from the store
const rockets = useSelector((state) => state.rockets)
You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/), a UI library that could speed up the process. This is a popular library and working with its components would be good practice.
Render a list of rockets (as per design). For the image of a rocket use the first image in the array of flickr_images.
Render a table with the missions' data (as per design).
Render a list of dragons (as per design). For the image of a dragon use the first image in the array of flickr_images[only if your team has 3 members].

First review:
Added tests for the project.
Add a filter for storing only needed attributes.

### CODE REVIEW

File: src/components/redux/missions/getMissions.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const url = ' https://api.spacexdata.com/v3/missions';
+const getMissions = createAsyncThunk('missions/getMissions', async () => {
+  try {
+    const response = await axios.get(url);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
+
```
Comment (TedLivist) [inline src/components/redux/missions/getMissions.js:13]: - [ ] Kindly filter the data to be saved from the API. It's not all items from the API that should be stored. Kindly visit [this requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store) and see the items to be saved.

File: src/components/redux/rockets/rocketsSlice.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import axios from 'axios';
+
+const baseUrl = 'https://api.spacexdata.com/v3/rockets';
+const initialState = {
+  rockets: [],
+  isLoading: false,
+  errorMessage: '',
+};
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', () => axios.get(baseUrl).then((response) => response.data));
```
Comment (TedLivist) [inline src/components/redux/rockets/rocketsSlice.js:11]: - [ ] Kindly filter the data to be saved from the API. It's not all items from the API that should be stored. Kindly visit [this requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#redux-fetch-data-and-update-redux-store) and see the items to be saved.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ghreza-crypto and @Cosmosha,

Good job so far!
✔️ Nice UI design
✔️ Dispatch of actions to store to reserve rockets and missions
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] Kindly add unit tests for your application. You have only one and it is failing.

   image

_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] **[OPTIONAL]** Kindly remove the `redux` from the `components` directory. The `redux` folder should be placed under `src`, not `src/components`

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @ghreza-crypto & @Cosmosha , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**